src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6919,6 +6920,230 @@ def create_stack(self, stack: StackRequest) -> StackResponse:
+                                    f"verified: {e}"
+                                )
+
+                            resources = connector_resources.resources[0]
+                            if resources.resource_ids:
+                                if len(resources.resource_ids) > 1:
+                                    raise RuntimeError(
+                                        f"Multiple {resource_type} resources are "
+                                        "available for the selected connector. "
+                                        "Please use the a specific resource-id to "
+                                        f"configure a {resource_type} resource."
+                                    )
+                                else:
+                                    resource_id = resources.resource_ids[0]
+
+                            component_update = ComponentUpdate(
+                                connector=service_connector.id,
+                                connector_resource_id=resource_id,
+                            )
+                            self.update_stack_component(
+                                component_id=component.id,
+                                component_update=component_update,
+                            )
+
+                    components_mapping[component_type] = [
+                        component.id,
+                    ]
+
+                # Stack
+                stack_name = full_stack.name
+                while True:
+                    try:
+                        stack_request = StackRequest(
+                            user=full_stack.user_id,
+                            workspace=full_stack.workspace_id,
+                            name=stack_name,
+                            description=full_stack.description,
+                            components=components_mapping,
+                        )
+                        stack_response = self.create_stack(stack_request)
+                        break
+                    except EntityExistsError:
+                        stack_name = (
+                            f"{full_stack.name}-{random_str(4)}".lower()
+                        )
+                        continue
+
+                handler.metadata.update(
+                    stack_response.get_analytics_metadata()
+                )
+            except Exception as e:
+                for service_connector_id in service_connectors_created_ids:
+                    self.delete_service_connector(
+                        service_connector_id=service_connector_id
+                    )
+                for component_id in components_created_ids:
+                    self.delete_stack_component(component_id=component_id)
+
+                raise RuntimeError(
+                    f"Full Stack creation has failed {e}. Cleaning up the "
+                    f"created entities."
+                )
```

Comment:
The real error is completely hidden from the logs. You should use `raise from` or `logger.exception()` here at least if debug logs are enabled. For example, this is what I got:

```
2024-06-28 18:37:23,876 - zenml.zen_server.utils - ERROR - API error (utils.py:457)
Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_stores/sql_zen_store.py", line 7015, in create_full_stack
    service_connector = service_connectors[
                        ^^^^^^^^^^^^^^^^^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_server/utils.py", line 447, in decorated
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_server/routers/workspaces_endpoints.py", line 417, in create_full_stack
    return zen_store().create_full_stack(full_stack)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_stores/sql_zen_store.py", line 7141, in create_full_stack
    raise RuntimeError(
RuntimeError: Full Stack creation has failed list index out of range. Cleaning up the created entities.
INFO:     3.78.248.73:0 - "POST /api/v1/workspaces/default/full-stack HTTP/1.1" 500 Internal Server Error
```

No idea where in the code this happened and which index is out of range.